test/test_optimizers.h

```diff
@@ -4,6 +4,106 @@
 
 namespace tiny_dnn {
 
+TEST(optimizers, adagrad_update) {
+    adagrad optimizer;
+
+    vec_t weights = {.2, .4, .006, -.77, -.010};
+    vec_t gradients = {1.0, -3.24, -.6, 2.79, 1.82};
+
+    // Defining the expected updates
+    vec_t first_update = {0.1900, 0.4100, 0.0160, -0.7800, -0.0200};
+    vec_t second_update = {0.1829, 0.4170, 0.0230, -0.7870, -0.0270};
+
+    // Testing
+    optimizer.update(gradients, weights, false);
+    for (auto i = 0u; i < weights.size(); i++)
```

Comment:
Just for the sake of consistency with other parts of code, could you change this to `size_t i = 0` ? Same for all for loops.